core/src/main/java/fr/sncf/osrd/envelope_sim_infra/MRSP.java

```diff
@@ -51,6 +51,8 @@ public static Envelope from(List<TrackRangeView> ranges, RollingStock rollingSto
                 var interval = speedRange.getKey();
                 var begin = offset + interval.getBeginPosition();
                 var end = offset + interval.getEndPosition();
+                if (addRollingStockLength)
+                    end += rollingStock.length;
```

Comment:
This change means that the MRSP envelope length could be larger than the path length. Same for any envelope built from the MRSP one. It doesn't have to be a problem, but it's something to keep in mind.